components/bio-formats-tools/src/loci/formats/tools/XMLValidate.java

```diff
@@ -47,17 +48,66 @@
  */
 public class XMLValidate {
 
-  public static void process(String label, BufferedReader in)
+  public static boolean validate(String label, BufferedReader in)
     throws IOException
   {
     StringBuffer sb = new StringBuffer();
-    while (true) {
-      String line = in.readLine();
-      if (line == null) break;
-      sb.append(line);
+    try {
+       while (true) {
+        String line = in.readLine();
+        if (line == null) break;
+        sb.append(line);
+      }
+    } finally {
+      in.close();
     }
-    in.close();
-    XMLTools.validateXML(sb.toString(), label);
+    return XMLTools.validateXML(sb.toString(), label);
+  }
+
+  @Deprecated
+  public static void process(String label, BufferedReader in)
+    throws IOException
+  {
+      validate(label, in);
+  }
+
+  public static boolean validate(String file)
+    throws IOException
+  {
+    String[] files = new String[1];
+    files[0] = file;
+    return validate(files)[0];
+  }
+
+  public static boolean[] validate(String[] files)
+    throws IOException
+  {
+    if (files == null || files.length == 0) {
+        throw new IllegalArgumentException("No files to validate");
+    }
+    boolean[] results = new boolean[files.length];
+    for (int i = 0; i < files.length; i++) {
+        String file = files[i];
+        if (file == null || file.trim().length() == 0) {
+          results[i] = false;
+        } else{
+          String f = file.toLowerCase();
+          boolean b;
+          if (f.endsWith("tif") || f.endsWith("tiff")) {
```

Comment:
RFE (outside the scope of this PR), we should extend the scope of this check and support all of `TiffReader.TIFF_SUFFIXES`

https://github.com/openmicroscopy/bioformats/blob/9fc607f85b8900be786813296f1eee75cc1ed883/components/formats-bsd/src/loci/formats/in/TiffReader.java#L73-L74